packages/elements/src/react/common/form/otp.tsx

```diff
@@ -88,173 +87,83 @@ const OTPInputStandard = React.forwardRef<HTMLInputElement, Omit<OTPInputProps,
  */
 const OTPInputSegmented = React.forwardRef<HTMLInputElement, Required<Pick<OTPInputProps, 'render'>> & OTPInputProps>(
   function OTPInput(props, ref) {
-    const {
-      className: userProvidedClassName,
-      render,
-      length = OTP_LENGTH_DEFAULT,
-      autoSubmit = false,
-      ...rest
-    } = props;
-
-    const value = props.value as string;
-    const previousValue = usePrevious(value);
+    const { className, render, length = OTP_LENGTH_DEFAULT, autoSubmit = false, ...rest } = props;
 
     const innerRef = React.useRef<HTMLInputElement>(null);
-    // This ensures we can access innerRef internally while still exposing it via the ref prop
-    React.useImperativeHandle(ref, () => innerRef.current as HTMLInputElement, []);
-
     const [selectionRange, setSelectionRange] = React.useState<SelectionRange>(props.autoFocus ? ZERO : OUTSIDE);
-    const isFocused = useFocus(innerRef);
+    const [isHovering, setIsHovering] = React.useState(false);
 
-    // Attach this component's UI state "selectionRange" to the native .select() to update it on focus changes
-    React.useEffect(() => {
-      const element = innerRef.current;
-
-      if (!element) return;
-
-      const originalSelect = element.select.bind(element);
-      element.select = () => {
-        originalSelect();
-        setSelectionRange([0, element.value.length]);
-      };
-    }, []);
-
-    // Continuously update the selection range when the input is focused
-    React.useEffect(() => {
-      if (!isFocused) return;
+    const isFocused = () => document.activeElement === innerRef.current;
```

Comment:
We should continue to use the `useFocus` hook since it covers more edge cases than this selector

Reply:
What edge cases?

Reply:
Ah, I forgot about SDK-1366 and that I haven't handled that yet. So right now the hook's only difference is checking for the existence of ref, otherwise it should behave the same. Once we use https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/interactions/src/useFocus.ts we can handle e.g. `iframe` edge cases